Documentation/Diagnostics/PH2019.md

```diff
@@ -0,0 +1,48 @@
+# PH2019: Avoid TestCleanup
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MsTestAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MsTestAnalyzers) |
+| Diagnostic ID | PH2019 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [AvoidAttributeAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MsTestAnalyzers/AvoidAttributeAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+The order that TestCleanup methods execute is not deterministic and can create unexpected test results. Moreover, it circumvents TestTimeouts.
+
+## How to solve
+
+Keep your test architecture simple. Remove the method. Rely on standard coding techniques to call the method at the beginning of your test.
+
+## Example
+
+Code that triggers a diagnostic:
+``` cs
+        [TestInitialize()]
+        public void Foo()
+        { }
+
+        [TestMethod]
+        public void BadTestMethod()
+        {
+            Assert.AreEqual(1, 1);
+        }
+```
+
+Fix the above code as follows:
+``` cs
+        [TestMethod]
+        public void BadTestMethod()
+        {
+            Assert.AreEqual(1, 1);
+            Foo();
```

Comment:
Consider renaming Foo to CleanUp for clarity

Reply:
done